src/services/user.service.js

```diff
@@ -0,0 +1,44 @@
+const { data } = require('../data');
+
+const getAll = () => {
+  return data.users;
+};
+
+const getById = (id) => {
+  return data.users.find((user) => user.id === id) || null;
+};
+
+const updateUser = ({ id, name }) => {
+  const user = getById(id);
+
+  Object.assign(user, { name });
+
+  return user;
+};
+
+const createUser = (name) => {
+  const user = {
+    name,
+    id: data.nextId++,
+  };
+
+  data.users.push(user);
+
+  return user;
```

Comment:
The `createUser` function is directly modifying the `data` array by pushing a new user into it. This could lead to unintended side effects in other parts of the code that are using the `data` array. Consider returning a new array with the new user added instead of modifying the existing one.